### DISCUSSION TITLE
fastn code for standard library stuff

### DISCUSSION THREAD

**Discussion by siddhantk232** [fastn-stack/fastn#1984] [Ideas & RFCs]
The current implementation of fastn has [this giant rust file](https://github.com/fastn-stack/fastn/blob/main/ftd/src/interpreter/things/default.rs) that defines most of the standard library stuff (kernel components etc).

We should be able to use `fastn` language. The language should provide constructs that support development of kernel components and target specialised components (component just for terminal, or android etc) in `fastn` language as much as possible.

This is a wild idea to explore so I am just gathering thoughts on this one for now.

**Comment by amitu** [comment]
A lot of the giant rust file can trivially be converted to fastn code, eg all the records / or types etc we define. This is clearly a good idea, and we should implement it in 0.5.

**Comment by Arpita-Jaiswal** [comment]
We have done some of this for `fastn` module imported in `FASTN.ftd` file where the `fastn.package`, `fastn.dependency` etc are defined in [`ftd` file](https://github.com/fastn-stack/fastn/blob/main/fastn-package/fastn_2021.ftd).
